woocommerce/wc-admin
# Add visual component docs from markdown examples

This PR is partially based off #358 – we set up a new page in the sidebar "DevDocs" to demo individual components. This only shows up if `WP_DEBUG && SCRIPT_DEBUG`. This renders the components in [`examples.json`](https://github.com/woocommerce/wc-admin/blob/update/docs-readme-examples/client/devdocs/examples.json), using the `example.md` file in each component folder.

List view shows only the demo:

![list](https://user-images.githubusercontent.com/541093/45906378-e437cc80-bdc1-11e8-931a-50d46800957f.png)

The single page view also grabs the documentation from `docs/components/${ component }.md`.

![single](https://user-images.githubusercontent.com/541093/45906382-e7cb5380-bdc1-11e8-822a-31b5da19b096.png)

Generally the examples are the same as what was in the "Usage" section of the README (now renamed to `example.md`, but many needed tweaks to actually run. 

Next steps after this PR (this one's getting too big):

- Add a section to the single view with the demo's source code, which will be fairly easy since it's markdown, we can use the same process as showing the component docs.
- Make sure this isn't being included in the production build – I'm pretty sure it is currently, even if the page isn't added.

**To test**

- Make sure you have `WP_DEBUG` and `SCRIPT_DEBUG` defined to true
- Open the DevDocs page from the sidebar, or go to `/wp-admin/admin.php?page=wc-admin#/devdocs`
- Play with the components. Most of the demos are functional, some are a little wonky (calendar, for example). Some look much better on the single page because they're complicated (like filters & table).
